pkg/controller/migmigration/registry.go

```diff
@@ -553,3 +557,121 @@ func (t *Task) getAnnotations(registryResources *MigRegistryResources) (map[stri
 	}
 	return annotations, nil
 }
+
+// Ensure that the migration registry is deleted on src and dest clusters after restore
+func (t *Task) ensureMigRegistryDelete() error {
+	// Delete dest Registry
+	err := t.ensureDestMigRegistryDelete()
+	if err != nil {
+		return err
+	}
+	// Delete src Registry
+	err = t.ensureSrcMigRegistryDelete()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Ensure that the migration registry is deleted on the src cluster after restore
+func (t *Task) ensureSrcMigRegistryDelete() error {
```

Comment:
The _ensure_ naming conventoin used in the controller implies create or update in the declarative sense.  IMHO both ensure & delete is a little confusing.  Also, I don't think there is a `MigRegistry` resource.  Is there?  Suggest something like: `deleteSrcRegistry()`

Reply:
Doesn't "delete" have the same problems as "create" -- what I really mean is "ensure that it's deleted" -- "delete" unless it's not there, in which case do nothing. In other words <the action of this method> is to "delete" as "ensure" is to "create". I'm not sure what the right word is. Ensure isn't quite right, as you point out. Delete doesn't seem to work either. In plain English, the action is "ensure deletion of", as opposed to "delete" because it is a declarative deletion.

Reply:
So we @djwhatle and I had the "MigRegistry" vs "MigrationRegistry" vs "Registry" naming discussion around my last PR. I'm trying to avoid just "Registry" to avoid any confusion with the internal registry. I originally had "MigrationRegistry" but @djwhatle was concerned that it was too long, so we ultimately agreed on "MigRegistry" -- in any case that's what is used throughout the creation code, so if we want to change it, it would need to be changed there too, if there's some other naming that makes more sense.

Reply:
> Doesn't "delete" have the same problems as "create" -- what I really mean is "ensure that it's deleted" -- "delete" unless it's not there, in which case do nothing.

Thanks for the explanation, @sseago.  You have convinced me.

Reply:
> So @djwhatle and I had the "MigRegistry" vs "MigrationRegistry" vs "Registry" naming discussion around my last PR. I'm trying to avoid just "Registry" to avoid any confusion with the internal registry.

I don't see any way this could be confused with the internal registry but find the `MigRegistry` confusing because it follows our CR naming convention.  That said, if you settled this with @djwhatle already, let's leave it for now.  It's not worth holding up this PR.